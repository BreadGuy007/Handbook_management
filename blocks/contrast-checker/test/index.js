/**
 * External dependencies
 */
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import ContrastChecker from '../';

describe( 'ContrastChecker', () => {
	const backgroundColor = '#ffffff';
	const textColor = '#000000';
	const isLargeText = true;
	const fallbackBackgroundColor = '#fff';
	const fallbackTextColor = '#000';
	const sameShade = '#666';

	const wrapper = shallow(
		<ContrastChecker
			backgroundColor={ backgroundColor }
			textColor={ textColor }
			isLargeText={ isLargeText }
			fallbackBackgroundColor={ fallbackBackgroundColor }
			fallbackTextColor={ fallbackTextColor } />
	);

	test( 'should render null when no colors are provided', () => {
		expect( shallow( <ContrastChecker /> ).html() ).toBeNull();
	} );

	test( 'should render null when the colors meet AA WCAG guidelines.', () => {
		expect( wrapper.html() ).toBeNull();
	} );

	test( 'should render component when the colors do not meet AA WCAG guidelines.', () => {
		const componentWrapper = shallow(
			<ContrastChecker
				backgroundColor={ sameShade }
				textColor={ sameShade }
				isLargeText={ isLargeText }
				fallbackBackgroundColor={ fallbackBackgroundColor }
				fallbackTextColor={ fallbackTextColor } />
		);

		expect( componentWrapper ).toMatchSnapshot();
	} );

	test( 'should render different message matching snapshot when background color has less brightness than text color.', () => {
		const darkerShade = '#555';

		const componentWrapper = shallow(
			<ContrastChecker
				backgroundColor={ darkerShade }
				textColor={ sameShade }
				isLargeText={ ! isLargeText }
				fallbackBackgroundColor={ fallbackBackgroundColor }
				fallbackTextColor={ fallbackTextColor } />
		);

		expect( componentWrapper ).toMatchSnapshot();
	} );

	test( 'should render null when the colors meet AA WCAG guidelines, with only fallback colors.', () => {
		const componentWrapper = shallow(
			<ContrastChecker
				isLargeText={ isLargeText }
				fallbackBackgroundColor={ fallbackBackgroundColor }
				fallbackTextColor={ fallbackTextColor } />
		);

		expect( componentWrapper.html() ).toBeNull();
	} );

	test( 'should render messages when the textColor is valid, but the fallback backgroundColor conflicts.', () => {
		const componentWrapper = shallow(
			<ContrastChecker
				textColor={ textColor }
				fallbackBackgroundColor={ textColor } />
		);

		expect( componentWrapper ).toMatchSnapshot();
	} );
} );
