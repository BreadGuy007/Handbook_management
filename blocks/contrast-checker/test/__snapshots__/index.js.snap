// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContrastChecker should render component when the colors do not meet AA WCAG guidelines. 1`] = `
<div
  className="blocks-contrast-checker"
>
  <Notice
    content="This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color."
    isDismissible={false}
    status="warning"
  />
</div>
`;

exports[`ContrastChecker should render different message matching snapshot when background color has less brightness than text color. 1`] = `
<div
  className="blocks-contrast-checker"
>
  <Notice
    content="This color combination may be hard for people to read. Try using a darker background color and/or a brighter text color."
    isDismissible={false}
    status="warning"
  />
</div>
`;

exports[`ContrastChecker should render messages when the textColor is valid, but the fallback backgroundColor conflicts. 1`] = `
<div
  className="blocks-contrast-checker"
>
  <Notice
    content="This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color."
    isDismissible={false}
    status="warning"
  />
</div>
`;
